frontend/lib/identification/user_code_store.dart

```diff
@@ -54,11 +41,31 @@ class UserCodeStore {
   // legacy import of existing card to keep them in storage after update
   Future<DynamicUserCode?> loadAndDeleteLegacyCard() async {
     return await _storage.use((storage) async {
-      final String? userCodeBase64 = await storage.read(key: _legacyUserCodeBase64Key);
+      final String? userCodeBase64 = await _safeRead(storage, _legacyUserCodeBase64Key);
       if (userCodeBase64 == null) return null;
       DynamicUserCode importedLegacyCard = DynamicUserCode.fromBuffer(const Base64Decoder().convert(userCodeBase64));
       await storage.delete(key: _legacyUserCodeBase64Key);
       return importedLegacyCard;
     });
   }
 }
+
+/// This function deletes the secure storage in the situation that the app data was backed up (e.g. on a new device),
+/// but the secret is no longer available.
+/// Upstream issue: https://github.com/juliansteenbakker/flutter_secure_storage/issues/354
+Future<String?> _safeRead(FlutterSecureStorage storage, String key) async {
+  try {
+    return await storage.read(key: key);
+  } on PlatformException catch (e, stackTrace) {
+    if (Platform.isAndroid && e.toString().contains('javax.crypto.BadPaddingException')) {
```

Comment:
The contains check might be a bit too unreliable. It is a bit weird that it throws a padding exception because the passing should remain valid actually.

Is there an issue with just deletingAll on any error?

Reply:
> It is a bit weird that it throws a padding exception because the passing should remain valid actually.

Yeah it's a weird error message; but it also says BAD_DECRYPT at some point. Maybe the padding somehow depends on the secret?

> Is there an issue with just deletingAll on any error?

I did not want to do that, as there might potentially be other errors and I am fearing data loss. If there are other errors, we at least get reported via Sentry and we can then include them here.